### PR TITLE
Add a generation field to plugins.

### PR DESCRIPTION
**Pull request by mdwn**
A new field has been added to the plugin object that will help uniquely identify the plugin and its current revision, which will be used to properly namespace Okta sourced access lists and resources.

Note: The fact that this will be missing on older plugins shouldn't be an issue, as its non-existence can be counted as a sort of uniqueness state.

### CODE REVIEW

**Comment by tcsc** [comment]
I think I'm missing something here: I'm not sure I understand how is this different to the `revision` field that already exists in the resource header that identifies a given revision of  the a resource record.

**Comment by camscale** [comment]
Could you not reformat all the comments in `types.proto`? It's pretty much unreviewable as I can't really tell what has changed there. If you do want to reformat the text, do that in one commit and put the actual changes in a second commit. But it does mess up the blame history if you do a large reformat like that so I'd leave it as is.

**Comment by mdwn** [comment]
> Could you not reformat all the comments in `types.proto`? It's pretty much unreviewable as I can't really tell what has changed there. If you do want to reformat the text, do that in one commit and put the actual changes in a second commit. But it does mess up the blame history if you do a large reformat like that so I'd leave it as is.

Argh! My IDE did this and I thought I fixed this. Will undo this.

**Comment by mdwn** [comment]
> I think I'm missing something here: I'm not sure I understand how is this different to the `revision` field that already exists in the resource header that identifies a given revision of  the a resource record.

@fspmarshall Has more context here, but he recommended against using this as the revision field isn't guaranteed to behave in the way that we need here. It could end up breaking Okta assignments unexpectedly.

**Comment by mdwn** [comment]
@camscale Fixed.

**Review by tcsc — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -5700,6 +5700,13 @@ message PluginSpecV1 {
     // Settings for the ServiceNow plugin
     PluginServiceNowSettings serviceNow = 10;
   }
+
+  // generation contains a unique ID that should:
+  // - Be created by the backed on plugin creation.
+  // - Be updated by the backend if the plugin is updated in any way.
+  //
+  // For older plugins, it's possible for this to be empty.
+  string generation = 11;
```
Comment (camscale) [inline api/proto/teleport/legacy/types/types.proto:5709]: Is this also an ordinal, as suggested by the name "generation" or just some random ID? If it is an ordinal, can you add that to the comment. Not sure whether a reader of this field should expect it to increase or just change when the generation changes.

**Review by camscale — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (mdwn) [reply api/proto/teleport/legacy/types/types.proto:5709]: I have this as a random ID. Just something random, really.

**Review by smallinsky — APPROVED** [review]

**Pull request merged**